cas-server-core-api-services/src/main/java/org/jasig/cas/services/RegisteredServiceAuthenticationPolicy.java

```diff
@@ -0,0 +1,44 @@
+package org.jasig.cas.services;
+
+import java.io.Serializable;
+import java.util.Set;
+
+/**
+ * This is {@link RegisteredServiceAuthenticationPolicy} that describes how a service
+ * should handle authentication requests.
+ *
+ * @author Misagh Moayyed
+ * @since 4.3.0
+ */
+public interface RegisteredServiceAuthenticationPolicy extends Serializable {
```

Comment:
If we do stick with a more generic name for the interface, we should rename the methods `isFailOpen()`, `getPrincipalAttributeNameTrigger()`, and `getPrincipalAttributeValueToMatch()` to make it clear they are MFA centric methods.